addons/website_sale/views/report_shop_saleorder.xml

```diff
@@ -15,10 +15,10 @@
                             <td>
                                 <strong>Payment Acquirer:</strong> <t t-esc="doc.payment_acquirer_id.name"/><br />
                                 <strong>Payment Status:</strong>
-                                <t t-if="doc.payment_acquirer_id.auto_confirm == 'none'">
+                                <t t-if="not doc.payment_acquirer_id.journal_id">
```

Comment:
Actually to remove as custom / transfer payments should be pending. Below code should handle it correctly. I think this t-if can be removed and the one just below does not need to check journal_id.